databricks/sdk/mixins/workspace.py

```diff
@@ -0,0 +1,40 @@
+import typing
+
+from ..service import workspace
+
+ExportFormat = workspace.ExportFormat
+Language = workspace.Language
+
+
+def _fqcn(x: any) -> str:
+    return f'{x.__module__}.{x.__name__}'
+
+
+class WorkspaceExt(workspace.WorkspaceAPI):
+
+    def upload(self,
+               path: str,
+               content: typing.BinaryIO,
+               *,
+               format: ExportFormat = ExportFormat.AUTO,
```

Comment:
This technically has different semantics than the Workspace API, which when the format is not specified then it will default to `format: AUTO`. 

Not opposed to this, but just want to make sure that this is a conscious choice.

Reply:
@jerryjam-db i've added path extension checking in the SDK, so that by default we can have this simple code to import the notebook without having to pass in the `language` parameter.

```python
    py = f'/Users/{w.current_user.me().user_name}/notebook-{random(12)}.py'

    w.workspace.upload(py, io.BytesIO(b'print(1)'))
    with w.workspace.download(py) as f:
        content = f.read()
        assert content == b'# Databricks notebook source\nprint(1)'
```

files are done via:

```python
    py_file = f'/Users/{w.current_user.me().user_name}/file-{random(12)}.py'

    w.workspace.upload(py_file, io.BytesIO(b'print(1)'), format=ExportFormat.AUTO)
    with w.workspace.download(py_file) as f:
        content = f.read()
        assert content == b'print(1)'

    w.workspace.delete(py_file)
```